### PR TITLE
vim-patch: update runtime files

### PR DESCRIPTION
**Pull request by clason**
- vim-patch:5eb9cb53d619 (omit vim9script autoload file)
- vim-patch:0f61943eb776
- vim-patch:82f19734bfcb (@dundargoc)

### CODE REVIEW

**Review by zeertzjq — APPROVED** [review]

File: runtime/indent/racket.vim [hunk 1]
```diff
@@ -3,26 +3,33 @@
...
 
 if exists("b:did_indent")
    finish
 endif
 let b:did_indent = 1
 
 setlocal lisp autoindent nosmartindent
+if has('vim9script')
+  setlocal indentexpr=racket#Indent() lispoptions+=expr:1
+endif
```
Comment (clason) [inline runtime/indent/racket.vim:16]: Upstream thought of us 🤗

**Review by zeertzjq — COMMENTED** [review]
Needs `make doc`

**Comment by clason** [comment]
Ah, I missed the meta annotations. (`gen_vimdoc.py` used to be enough, which I did.)

**Review by zeertzjq — APPROVED** [review]

File: runtime/doc/options.txt [hunk 2]
```diff
@@ -2831,9 +2831,9 @@ A jump table for the options with a short description can be found at |Q_op|.
 		:s///g		  subst. one	  subst. all
 		:s///gg		  subst. all	  subst. one
 
-	DEPRECATED: Setting this option may break plugins that are not aware
-	of this option.  Also, many users get confused that adding the /g flag
-	has the opposite effect of that it normally does.
+	NOTE: Setting this option may break plugins that rely on the default
+	behavior of the 'g' flag. This will also make the 'g' flag have the
+	opposite effect of that documented in |:s_g|.
```
Comment (gpanders) [inline runtime/doc/options.txt:2837]: Should `'gdefault'` also be removed from `deprecated.txt`?

Comment (clason) [reply runtime/doc/options.txt:2837]: Probably; can be smooshed into the collected docs PR? @dundargoc

**Review by seandewar — APPROVED** [review]

File: runtime/doc/options.txt (hunk 2 above)
Comment (dundargoc) [reply runtime/doc/options.txt:2837]: Sure

**Pull request merged**